platform/viewer/cypress/integration/common/OHIFDownloadSnapshotFile.spec.js

```diff
@@ -0,0 +1,152 @@
+describe('OHIF Download Snapshot File', () => {
+  before(() => {
+    cy.openStudy('MISTER^MR');
+    cy.waitDicomImage();
+    cy.expectMinimumThumbnails(5);
+  });
+
+  beforeEach(() => {
+    cy.openDownloadImageModal();
+  });
+
+  afterEach(() => {
+    cy.get('[data-cy="close-button"]')
+      .scrollIntoView()
+      .click();
+  });
+
+  it('checks displayed information for Tablet experience', function() {
+    // Set Tablet resolution
+    cy.viewport(1000, 660);
+    // Visual comparison
+    cy.screenshot('Download Image Modal - Tablet experience');
+    cy.percyCanvasSnapshot('Download Image Modal - Tablet experience');
+  });
+
+  it('checks displayed information for Desktop experience', function() {
+    // Set Desktop resolution
+    cy.viewport(1750, 720);
+    // Visual comparison
+    cy.screenshot('Download Image Modal - Desktop experience');
+    cy.percyCanvasSnapshot('Download Image Modal - Desktop experience');
+    //Check if all elements are displayed
+    cy.get('[data-cy=OHIFModal-header]')
+      .as('downloadImageModal')
+      .should('contain.text', 'Download High Quality Image');
+    // Check input fields
+    cy.get('[data-cy=file-info-container]')
+      .should('contain.text', 'Image width (px)')
+      .should('contain.text', 'Image height (px)')
+      .and('contain.text', 'File name')
+      .and('contain.text', 'File type')
+      .and('contain.text', 'Show Annotations');
+    cy.get('[data-cy=file-type]')
+      .select('png')
+      .should('have.value', 'png')
+      .select('jpg')
+      .should('have.value', 'jpg');
+    // Check image preview
+    cy.get('[data-cy="image-preview"]').should('contain.text', 'Image Preview');
+    cy.get('[data-cy="viewport-preview-img"]')
+      .should('have.attr', 'src')
+      .and('include', 'data:image');
+    // Check buttons
+    cy.get('[data-cy="cancel-btn"]')
+      .scrollIntoView()
+      .should('have.text', 'Cancel');
```

Comment:
I don't think we care if it says `Cancel`. We probably mostly care that clicking it has the expected action occur.

Reply:
The idea is to make sure all elements are displayed, but we can check the same thing with Percy Snapshot comparison. The action of closing the modal is being tested here: `cancel changes on download modal`. Having that said, we can remove this.

Reply:
> The idea is to make sure all elements are displayed

I think you've hit the nail on the head when you say:

> we can check the same thing with Percy Snapshot comparison

Visual testing would be too much to maintain with Cypress alone, and very brittle. Percy does _just enough_ to give us a gut check without a huge maintenance overhead.